build.gradle

```diff
@@ -108,8 +104,42 @@ apply from: 'deploy.gradle'
 apply from: 'tasks.gradle'
 apply from: 'jmh.gradle'
 
+project(':iceberg-relocate-external-dependencies') {
+  apply plugin: 'com.github.johnrengelman.shadow'
+  
+  tasks.javadoc.enabled = false
+  tasks.javadocJar.enabled = false
+  tasks.assemble.dependsOn tasks.shadowJar
+  tasks.install.dependsOn tasks.shadowJar
+  
+  dependencies {    
+    compile('com.google.guava:guava:28.0-jre') {
+      exclude group: 'com.google.code.findbugs'
+      // may be LGPL - use ALv2 findbugs-annotations instead
+      exclude group: 'com.google.errorprone'
+      exclude group: 'com.google.j2objc'
+    }
+  }
+  
+  shadowJar {
+    configurations = [project.configurations.compile]
+    zip64 true
+
+    // include the LICENSE and NOTICE files for the shaded Jar
+    from(projectDir) {
+      include 'LICENSE'
+      include 'NOTICE'
+    }
+
+    relocate 'com.google.common', 'com.google.common.shaded'
+    
+    minimize()
+  }
+}
+
 project(':iceberg-api') {
   dependencies {
+    compile project(path: ':iceberg-relocate-external-dependencies', configuration: 'shadow')
```

Comment:
Does this work well with your IDE?

Reply:
No, but I'm using Eclipse and all kinds of things don't work, haven't had time to go through them all and figure them out. Can you try it on your side? It works on CLI so I was assuming it would be OK in an IDE but perhaps not?

Reply:
I've got it working with IntelliJ by building the `relocate-external-dependencies` first